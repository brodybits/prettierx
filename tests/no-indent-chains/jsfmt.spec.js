run_spec(__dirname, ["babel", "babel-flow", "flow", "typescript"], {
  indentChains: false
});

run_spec(__dirname, ["babel", "babel-flow", "flow", "typescript"], {
  indentChains: false,
  parenSpacing: true
});
