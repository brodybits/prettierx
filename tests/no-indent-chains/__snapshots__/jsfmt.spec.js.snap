// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`member-chain.js 1`] = `
====================================options=====================================
indentChains: false
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
foo.doSomething('Hello World').doAnotherThing('Foo', { foo: bar })
  // App configuration.
  .doOneMoreThing(config)
  .run(() => console.log('Bar'));

bigDeal
  .doSomething('Hello World')

  // Hello world
  .doAnotherThing('Foo', { foo: bar })

  // App configuration.
  .doOneMoreThing(config)

  .run(() => console.log('Bar'));


bar.baz
  .doSomething('Hello World')

  // Hello world
  .foo.bar.doAnotherThing('Foo', { foo: bar })

  .doOneMoreThing(config)
  .bar.run(() => console.log('Bar'));

(
  somethingGood ? thisIsIt : maybeNot
)
  // Hello world
  .doSomething('Hello World')
  .doAnotherThing('Foo', { foo: bar }) // Run this
  .run(() => console.log('Bar')); // Do this

helloWorld
  .text()
  .then(t => t);

(veryLongVeryLongVeryLong ||
 anotherVeryLongVeryLongVeryLong ||
 veryVeryVeryLongError
)
  .map(tickets => TicketRecord.createFromSomeLongString())

  .filter(obj => !!obj);

const sel = this.connections
  .concat(this.activities.concat(this.operators))
  .filter(x => x.selected);

=====================================output=====================================
foo
.doSomething("Hello World")
.doAnotherThing("Foo", { foo: bar })
// App configuration.
.doOneMoreThing(config)
.run(() => console.log("Bar"));

bigDeal
.doSomething("Hello World")

// Hello world
.doAnotherThing("Foo", { foo: bar })

// App configuration.
.doOneMoreThing(config)

.run(() => console.log("Bar"));

bar.baz
.doSomething("Hello World")

// Hello world
.foo.bar.doAnotherThing("Foo", { foo: bar })

.doOneMoreThing(config)
.bar.run(() => console.log("Bar"));

(somethingGood ? thisIsIt : maybeNot)
// Hello world
.doSomething("Hello World")
.doAnotherThing("Foo", { foo: bar }) // Run this
.run(() => console.log("Bar")); // Do this

helloWorld.text().then(t => t);

(
  veryLongVeryLongVeryLong ||
  anotherVeryLongVeryLongVeryLong ||
  veryVeryVeryLongError
)
.map(tickets => TicketRecord.createFromSomeLongString())

.filter(obj => !!obj);

const sel = this.connections
.concat(this.activities.concat(this.operators))
.filter(x => x.selected);

================================================================================
`;

exports[`member-chain.js 2`] = `
====================================options=====================================
indentChains: false
parenSpacing: true
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
foo.doSomething('Hello World').doAnotherThing('Foo', { foo: bar })
  // App configuration.
  .doOneMoreThing(config)
  .run(() => console.log('Bar'));

bigDeal
  .doSomething('Hello World')

  // Hello world
  .doAnotherThing('Foo', { foo: bar })

  // App configuration.
  .doOneMoreThing(config)

  .run(() => console.log('Bar'));


bar.baz
  .doSomething('Hello World')

  // Hello world
  .foo.bar.doAnotherThing('Foo', { foo: bar })

  .doOneMoreThing(config)
  .bar.run(() => console.log('Bar'));

(
  somethingGood ? thisIsIt : maybeNot
)
  // Hello world
  .doSomething('Hello World')
  .doAnotherThing('Foo', { foo: bar }) // Run this
  .run(() => console.log('Bar')); // Do this

helloWorld
  .text()
  .then(t => t);

(veryLongVeryLongVeryLong ||
 anotherVeryLongVeryLongVeryLong ||
 veryVeryVeryLongError
)
  .map(tickets => TicketRecord.createFromSomeLongString())

  .filter(obj => !!obj);

const sel = this.connections
  .concat(this.activities.concat(this.operators))
  .filter(x => x.selected);

=====================================output=====================================
foo
.doSomething( "Hello World" )
.doAnotherThing( "Foo", { foo: bar } )
// App configuration.
.doOneMoreThing( config )
.run( () => console.log( "Bar" ) );

bigDeal
.doSomething( "Hello World" )

// Hello world
.doAnotherThing( "Foo", { foo: bar } )

// App configuration.
.doOneMoreThing( config )

.run( () => console.log( "Bar" ) );

bar.baz
.doSomething( "Hello World" )

// Hello world
.foo.bar.doAnotherThing( "Foo", { foo: bar } )

.doOneMoreThing( config )
.bar.run( () => console.log( "Bar" ) );

( somethingGood ? thisIsIt : maybeNot )
// Hello world
.doSomething( "Hello World" )
.doAnotherThing( "Foo", { foo: bar } ) // Run this
.run( () => console.log( "Bar" ) ); // Do this

helloWorld.text().then( t => t );

(
  veryLongVeryLongVeryLong ||
  anotherVeryLongVeryLongVeryLong ||
  veryVeryVeryLongError
)
.map( tickets => TicketRecord.createFromSomeLongString() )

.filter( obj => !! obj );

const sel = this.connections
.concat( this.activities.concat( this.operators ) )
.filter( x => x.selected );

================================================================================
`;
